language/src/typechecker.rs

```diff
@@ -2220,8 +2382,109 @@ fn typecheck_question_mark(
         dealias_type(return_typ.0.clone(), top_level_context),
         return_typ.1.clone(),
     );
+
     if question_mark {
         let expr_carrier: Result<CarrierTyp, _> = match expr_typ.clone() {
+            (
```

Comment:
I changed this to do unification, I had an issue with AES not working without this.